crates/oxc_prettier/src/format/mod.rs

```diff
@@ -860,7 +869,45 @@ impl<'a> Format<'a> for TSLiteralType<'a> {
 
 impl<'a> Format<'a> for TSMappedType<'a> {
     fn format(&self, p: &mut Prettier<'a>) -> Doc<'a> {
-        line!()
+        let mut parts: Vec<'_, Doc<'_>> = p.vec();
+
+        match self.readonly {
+            TSMappedTypeModifierOperator::Plus => parts.push(ss!("+readonly ")),
+            TSMappedTypeModifierOperator::Minus => parts.push(ss!("-readonly ")),
+            TSMappedTypeModifierOperator::True => parts.push(ss!("readonly ")),
+            TSMappedTypeModifierOperator::None => (),
+        }
+
+        parts.push(ss!("["));
+        parts.push(self.type_parameter.format(p));
+
+        if let Some(name_type) = &self.name_type {
+            parts.push(ss!(" as "));
+            parts.push(name_type.format(p));
+        }
+
+        parts.push(ss!("]"));
+
+        match self.optional {
+            TSMappedTypeModifierOperator::Plus => parts.push(ss!("+?")),
```

Comment:
Could we add an `as_str` method to `TSMappedTypeModifierOutput` in `oxc_ast`?

Reply:
Should I use for  `TSMappedTypeModifierOperator::None`  an empty string? 
Is it okay to push empty string or should I check?

Reply:
Good find, ignore this comment then!